pkg/pillar/hypervisor/xen.go

```diff
@@ -583,6 +580,44 @@ func (ctx xenContext) PCIRelease(long string) error {
 	return nil
 }
 
+//IsDomainHealthy: Verifies if a domain is up and running by its state. Domain can have the following states:
```

Comment:
This still doesn't explain what it returns when it can't determine whether it is healthy or not. That indeterminate state is the key source of the complexity hence the comment must explain it.

Reply:
I suggest adding "When we can not determine whether it is healthy e.g., due to a pending state transition, we return false. But the caller must not assume this means it is failed since it might be transitioning between healhy states."

And based on that I'd rename the function from "Healhy" to "MaybeHealthy" to reduce the risk of confusion.

Reply:
Changed the comment and method name.